src/python/pants/backend/java/target_types.py

```diff
@@ -18,40 +18,68 @@
 from pants.engine.unions import UnionMembership, UnionRule
 
 
-class JavaSources(Sources):
+class JavaSourceField(Sources):
     expected_file_extensions = (".java",)
+    expected_num_files = 1
+
+
+class JavaGeneratorSources(Sources):
+    pass
+    # TODO: do I need anything else here?
+
+
+# -----------------------------------------------------------------------------------------------
+# `java_test` target
```

Comment:
You're using Java in some places, junit in others. I don't have a strong opinion what we should use. I think it depends on how we're going to model Scala tests.

For now, feel free to choose either, so long as it's consistent.

Reply:
I'm sticking with JUnit until we discuss how (if) we want to deal with different testing frameworks. This is similar to how we talk about `pytest` over in Pythonland

Reply:
I imagine that we will end up keeping the test runner's name in the name of the target type. E.g., for ScalaTest, it will probably be called `scalatest_test` (notwithstanding dealing with the duplication of "test" in the name of the framework and as the name of what the target is).